### DISCUSSION TITLE
Ability to have other enum rawValue parser

### DISCUSSION THREAD

**Discussion by the-yuyut** [pointfreeco/swift-parsing#276] [General]
Hi, I would like to have my enum be a Parse. I have learned that we can use `RawValue` and `CaseIterable` on the enum type to gain the Parse for free. 

```swift
enum Person: Int, CaseIterable {
      case blob = 4
      case blobJr = 42
}
```

For some reason, I would like to have my Parser adopt UInt8 enum, But I found that it is not achievable. 

```swift
enum Format: UInt8, CaseIterable {
      case ascii = 0x1
      case unicode = 0x2
}
``` 

> For any enumeration with a string, integer, or floating-point raw type, the Swift compiler automatically adds RawRepresentable conformance. 
>  - [RawRepresentable | Apple Developer Documentation](https://developer.apple.com/documentation/swift/rawrepresentable), 

IMO, I have simple way for me is to extend `CaseIterable && RawRepresentable && UInt8`, like  `CaseIterableRawRepresentable.swift` in a3d52c666982f7cc9f243b910151cdb9f6283db1

https://github.com/pointfreeco/swift-parsing/blob/a3d52c666982f7cc9f243b910151cdb9f6283db1/Sources/Parsing/ParserPrinters/CaseIterableRawRepresentable.swift#L1-L16

Then, I found that `Parsers.CaseIterableRawRepresentableParser` is public, but its `init` is `internal` 😞

**Therefore, I would like to propose making `Parsers.CaseIterableRawRepresentableParser.init` public!**

The patch is quite simple as follows.

```diff
diff --git a/Sources/Parsing/ParserPrinters/CaseIterableRawRepresentable.swift b/Sources/Parsing/ParserPrinters/CaseIterableRawRepresentable.swift
index f401edd5a..3336cac0e 100644
--- a/Sources/Parsing/ParserPrinters/CaseIterableRawRepresentable.swift
+++ b/Sources/Parsing/ParserPrinters/CaseIterableRawRepresentable.swift
@@ -118,8 +118,7 @@ extension Parsers {
     @usableFromInline
     let cases: [(case: Output, prefix: Prefix, count: Int)]
 
-    @usableFromInline
-    init(
+    public init(
       toPrefix: @escaping (Output.RawValue) -> Prefix,
       areEquivalent: @escaping (Input.Element, Input.Element) -> Bool
     ) {

```

**Comment by mbrandonw** [comment]
Hi @ytyubox, good catch! 

Rather than making the initializer public, what if we made the parser work with all `FixedWidthInteger` types? Then we could all the various integer types.

If you have the time you want to open a PR to do that? If not we can take care of it.

**Reply by the-yuyut** [reply]
Sure, here you go.https://github.com/pointfreeco/swift-parsing/pull/277

**Reply by the-yuyut** [reply]
Hi, @mbrandonw can you please take a look?

**Reply by mbrandonw** [reply]
Hi @ytyubox, thanks for following up. Just merged!
